build.go

```diff
@@ -189,46 +182,76 @@ func (p spkParams) outputFile() string {
 }
 
 func (p spkParams) goEnv() []string {
-	return append(os.Environ(),
+	goarch := p.goarch
+	var goarm string
+	if strings.HasPrefix(goarch, "arm/") {
+		goarm = goarch[len(goarch)-1:]
+		goarch = "arm"
+	}
+	env := append(os.Environ(),
+		"CGO_ENABLED=0",
 		"GOOS=linux",
-		"GOARCH="+p.goarch,
-		// TODO: add GOARM, if we ever start building GOARM variants
+		"GOARCH="+goarch,
 	)
-}
-
-// genSPK generates an SPK, which is a nested tarball.
-// The outer tar file is uncompressed and contains the minimal
-// metadata. The main contents are in the outer tar's "package.tgz"
-// entry, which is gzip or xz compressed.
-func genSPK(w io.Writer, param spkParams) error {
-	privFile := fmt.Sprintf("privilege-dsm%d", param.dsm)
-	if param.forPackageCenter {
-		privFile += ".priv"
+	if goarm != "" {
+		env = append(env, "GOARM="+goarm)
 	}
+	return env
+}
 
-	var innerPkgTgz bytes.Buffer
-	extractedSize, err := genInnerPackageTgz(&innerPkgTgz, param)
+func genSPK(param spkParams, dsm int, synoArch string, extractedSize int64, privFile string, innerPackage []byte) error {
+	out := param.outputFile(dsm, synoArch)
+	log.Printf("Generating %v ...", filepath.Base(out))
+	w, err := os.OpenFile(out, os.O_CREATE|os.O_WRONLY, 0644)
 	if err != nil {
 		return err
 	}
-
+	defer w.Close()
 	return writeTar(w,
-		file("INFO", getInfo(param, extractedSize)),
+		file("INFO", getInfo(param, dsm, synoArch, extractedSize)),
 		file("PACKAGE_ICON.PNG", static("PACKAGE_ICON.PNG")),
 		file("PACKAGE_ICON_256.PNG", static("PACKAGE_ICON_256.PNG")),
 		file("Tailscale.sc", static("Tailscale.sc")),
 		dir("conf/", param.createTime),
 		file("conf/resource", static("resource")),
 		file("conf/privilege", static(privFile)),
-		file("package.tgz", memFile(innerPkgTgz.Bytes(), 0644, param.createTime)),
+		file("package.tgz", memFile(innerPackage, 0644, param.createTime)),
 		dir("scripts/", param.createTime),
 		file("scripts/start-stop-status", static("scripts/start-stop-status")),
 		file("scripts/postupgrade", static("scripts/postupgrade")),
 		file("scripts/preupgrade", static("scripts/preupgrade")),
 	)
 }
 
-func genInnerPackageTgz(w io.Writer, param spkParams) (extractedSize int64, err error) {
+// genArchSPKs generates SPKs for a particular DSM version.
+// This function generates one spk per synoArch.
+// SPKs are nested tarballs.
+// The outer tar file is uncompressed and contains the minimal
+// metadata. The main contents are in the outer tar's "package.tgz"
+// entry, which is gzip or xz compressed.
+func genArchSPKs(param spkParams, dsm int, synoArchs []string) error {
+	var innerPkgTgz bytes.Buffer
```

Comment:
I like how this is now shared between all the genSPKs for different synoarch names!